jsonstreamer/jsonstreamer.py

```diff
@@ -146,7 +146,7 @@ def on_double(self, ctx, value):
 
     def on_number(self, ctx, value):
         ''' Since this is defined both integer and double callbacks are useless '''
-        value = float(value) if '.' in value else int(value)
+        value = int(value) if value.isdigit() else float(value)
```

Comment:
Yes, like '9e-06' but only floats/numbers.
For strings we have a different listener (on_string)